drivers/hmis/app/models/hmis/file.rb

```diff
@@ -35,11 +35,24 @@ def data_source
   scope :nonconfidential, -> { where(confidential: [false, nil]) }
 
   scope :viewable_by, ->(user) do
-    view_scope = where(client_id: Hmis::Hud::Client.with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files))
-    # view_scope = view_scope.nonconfidential unless user.can_view_any_confidential_client_files?
-    edit_scope = user.can_manage_own_client_files? ? where(user_id: user.id) : none
+    client_scope = Hmis::Hud::Client
+      .with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files)
+    enrollment_scope = Hmis::Hud::Enrollment
+      .with_access(user, :can_view_any_nonconfidential_client_files, :can_view_any_confidential_client_files)
```

Comment:
```suggestion
    client_scope = Hmis::Hud:: Enrollment.viewable_by(user)
```

We need this in addition to the with_access check because it's possible that the user might be assigned "can_view_any_confidential_client_files" for some project, but not have the requisite other permissions that are needed to actually view that enrollment ("can_view_project" + "can_view_enrollment_details"). There is a comment about this on the `with_access` scope